osu.Game.Tests/Database/RealmLiveTests.cs

```diff
@@ -38,22 +38,22 @@ public void TestAccessAfterStorageMigrate()
             {
                 var beatmap = new BeatmapInfo(CreateRuleset(), new BeatmapDifficulty(), new BeatmapMetadata());
 
-                ILive<BeatmapInfo> liveBeatmap;
+                ILive<BeatmapInfo>? liveBeatmap = null;
 
-                using (var context = realmFactory.CreateContext())
+                realmFactory.Run(realm =>
                 {
-                    context.Write(r => r.Add(beatmap));
+                    realm.Write(r => r.Add(beatmap));
 
                     liveBeatmap = beatmap.ToLive(realmFactory);
-                }
+                });
```

Comment:
This test is failing on windows on CI and I suspect this change is the cause. I haven't confirmed experimentally, but my educated guess is that the old test implicitly relied on explicit creation of disposal of a realm context, even though this code was running on the update thread. That was the only realm context used, which means that when it was disposed nothing used realm anymore.

Now with the change to `Run()`, this will use `RealmContextFactory.Context`,  which will not be disposed until the factory is (or until a `BlockAllOperations()` call), causing failures on windows only because it is seems to be the only OS on which the realm file is locked even for reads if any context is open.

Reply:
Has been fixed by triggering a recycle 👍